frontend/packages/dev-console/src/components/import/GitImport.tsx

```diff
@@ -72,26 +66,17 @@ const GitImport: React.FC<GitImportProps> = ({ namespace, imageStreams }) => {
 
     const {
       project: { name: projectName },
-      route: { create: canCreateRoute },
-      image: { ports },
     } = values;
 
-    const requests = [
-      createDeploymentConfig(values, imageStream),
-      createImageStream(values, imageStream),
-      createBuildConfig(values, imageStream),
-    ];
+    const dryRunRequests: Promise<K8sResourceKind>[] = createResources(values, imageStream, {
+      queryParams: { dryRun: 'All' },
+    });
 
-    // Only create a service or route if the builder image has ports.
-    if (!_.isEmpty(ports)) {
-      requests.push(createService(values, imageStream));
-      if (canCreateRoute) {
-        requests.push(createRoute(values, imageStream));
-      }
-    }
-
-    requests.forEach((r) => r.catch((err) => actions.setStatus({ submitError: err.message })));
-    Promise.all(requests)
+    Promise.all(dryRunRequests)
+      .then(() => {
+        const requests: Promise<K8sResourceKind>[] = createResources(values, imageStream);
```

Comment:
We should add some kind of loading state or message to the form that indicates that its doing dry run so that user doesn't get the idea that the form in unresponsive at any time.

Reply:
This isn't specific to dry run state but even creation _could_ take some time.
We can handle this extra request in functionality as a separate PR.